test/integration/integration.test.ts

```diff
@@ -0,0 +1,533 @@
+          data: expect.objectContaining({
+            walletAddressUrl: notFoundWalletAddress
+          })
+        })
+      )
+    })
+
+    test('Grant Request Incoming Payment', async (): Promise<void> => {
+      const grant = await c9.opClient.grant.request(
+        {
+          url: receiverWalletAddress.authServer
+        },
+        {
+          access_token: {
+            access: [
+              {
+                type: 'incoming-payment',
+                actions: ['create', 'read', 'list', 'complete']
+              }
+            ]
+          }
+        }
+      )
+
+      assert(!isPendingGrant(grant))
+      accessToken = grant.access_token.value
+    })
+
+    test('Create Incoming Payment', async (): Promise<void> => {
+      const now = new Date()
+      const tomorrow = new Date(now)
+      tomorrow.setDate(now.getDate() + 1)
+
+      const handleWebhookEventSpy = jest.spyOn(
+        hlb.integrationServer.webhookEventHandler,
+        'handleWebhookEvent'
+      )
+
+      incomingPayment = await c9.opClient.incomingPayment.create(
+        {
+          url: receiverWalletAddress.resourceServer,
+          accessToken
+        },
+        {
+          walletAddress: receiverWalletAddressUrl.replace('http', 'https'),
+          incomingAmount: {
+            value: '100',
+            assetCode: receiverWalletAddress.assetCode,
+            assetScale: receiverWalletAddress.assetScale
+          },
+          metadata: { description: 'Free Money!' },
+          expiresAt: tomorrow.toISOString()
+        }
+      )
+
+      // Delay gives time for webhook to be received
+      await wait(1000)
+      expect(handleWebhookEventSpy).toHaveBeenCalledWith(
+        expect.objectContaining({
+          type: WebhookEventType.IncomingPaymentCreated,
+          data: expect.any(Object)
+        })
+      )
+    })
+
+    test('Grant Request Quote', async (): Promise<void> => {
+      const grant = await c9.opClient.grant.request(
+        {
+          url: senderWalletAddress.authServer
+        },
+        {
+          access_token: {
+            access: [
+              {
+                type: 'quote',
+                actions: ['read', 'create']
+              }
+            ]
+          }
+        }
+      )
+
+      assert(!isPendingGrant(grant))
+      accessToken = grant.access_token.value
+    })
+
+    test('Create Quote', async (): Promise<void> => {
+      quote = await c9.opClient.quote.create(
+        {
+          url: senderWalletAddress.resourceServer,
+          accessToken
+        },
+        {
+          walletAddress: senderWalletAddressUrl.replace('http', 'https'),
+          receiver: incomingPayment.id.replace('https', 'http'),
+          method: 'ilp'
+        }
+      )
+    })
+
+    // --- GRANT CONTINUATION WITH FINISH METHOD ---
+    // TODO: Grant Continuation w/ finish in another Open Payments Flow test
+    test.skip('Grant Request Outgoing Payment', async (): Promise<void> => {
```

Comment:
Currently the "Open Payments Flow" uses continuation polling but we can test both. The only thing stopping me ATM is that we'd have to duplicate all the other test steps, which can be improved and is captured here: https://github.com/interledger/rafiki/issues/2529 So I'm just leaving these alternate steps for now